src/ucm/cuda/cudamem.c

```diff
@@ -154,23 +179,42 @@ UCM_CUDA_ALLOC_FUNC(cuMemAllocPitch_v2, UCS_MEMORY_TYPE_CUDA, CUresult,
                     CUDA_SUCCESS, (size_t)arg1 * arg2, CUdeviceptr,
                     "pitch=%p width=%zu height=%zu elem=%u", size_t*, size_t,
                     size_t, unsigned)
+#if CUDA_VERSION >= 11020
+UCM_CUDA_ALLOC_FUNC(cuMemAllocAsync, UCS_MEMORY_TYPE_CUDA_MANAGED, CUresult,
+                    CUDA_SUCCESS, arg0, CUdeviceptr, "size=%zu stream=%p",
+                    size_t, CUstream)
+UCM_CUDA_ALLOC_FUNC(cuMemAllocFromPoolAsync, UCS_MEMORY_TYPE_CUDA_MANAGED,
+                    CUresult, CUDA_SUCCESS, arg0, CUdeviceptr,
+                    "size=%zu pool=%p stream=%p", size_t, CUmemoryPool,
+                    CUstream)
+#endif
 UCM_CUDA_FREE_FUNC(cuMemFree, CUresult, CUdeviceptr, UCS_MEMORY_TYPE_CUDA)
 UCM_CUDA_FREE_FUNC(cuMemFree_v2, CUresult, CUdeviceptr, UCS_MEMORY_TYPE_CUDA)
 UCM_CUDA_FREE_FUNC(cuMemFreeHost, CUresult, void*, UCS_MEMORY_TYPE_HOST)
 UCM_CUDA_FREE_FUNC(cuMemFreeHost_v2, CUresult, void*, UCS_MEMORY_TYPE_HOST)
+#if CUDA_VERSION >= 11020
+UCM_CUDA_FREE_ASYNC_FUNC(cuMemFreeAsync, CUresult, CUdeviceptr, CUstream,
+                         UCS_MEMORY_TYPE_CUDA_MANAGED)
+#endif
 
-static ucm_cuda_func_t ucm_cuda_driver_funcs[] = {
-    UCM_CUDA_FUNC_ENTRY(cuMemAlloc),
-    UCM_CUDA_FUNC_ENTRY(cuMemAlloc_v2),
-    UCM_CUDA_FUNC_ENTRY(cuMemAllocManaged),
-    UCM_CUDA_FUNC_ENTRY(cuMemAllocPitch),
-    UCM_CUDA_FUNC_ENTRY(cuMemAllocPitch_v2),
-    UCM_CUDA_FUNC_ENTRY(cuMemFree),
-    UCM_CUDA_FUNC_ENTRY(cuMemFree_v2),
-    UCM_CUDA_FUNC_ENTRY(cuMemFreeHost),
-    UCM_CUDA_FUNC_ENTRY(cuMemFreeHost_v2),
-    {{NULL}, NULL}
-};
+static ucm_cuda_func_t ucm_cuda_driver_funcs[] =
+        {UCM_CUDA_FUNC_ENTRY(cuMemAlloc),
+         UCM_CUDA_FUNC_ENTRY(cuMemAlloc_v2),
+         UCM_CUDA_FUNC_ENTRY(cuMemAllocManaged),
+         UCM_CUDA_FUNC_ENTRY(cuMemAllocPitch),
+         UCM_CUDA_FUNC_ENTRY(cuMemAllocPitch_v2),
+#if CUDA_VERSION >= 11020
+         UCM_CUDA_FUNC_ENTRY(cuMemAllocAsync),
+         UCM_CUDA_FUNC_ENTRY(cuMemAllocFromPoolAsync),
+#endif
+         UCM_CUDA_FUNC_ENTRY(cuMemFree),
+         UCM_CUDA_FUNC_ENTRY(cuMemFree_v2),
+         UCM_CUDA_FUNC_ENTRY(cuMemFreeHost),
+         UCM_CUDA_FUNC_ENTRY(cuMemFreeHost_v2),
+#if CUDA_VERSION >= 11020
+         UCM_CUDA_FUNC_ENTRY(cuMemFreeAsync),
+#endif
+         {{NULL}, NULL}};
```

Comment:
for completeness, can you add hooks below for runtime APIs as well? it might useful in when bistro hooks fails

Reply:
Done.